src/Previewer/Internal.js

```diff
@@ -64,14 +65,48 @@ export default function Internal({
 
   function updateBuilderRoot(object_preview) {
     stack.set_visible_child_name("open_window");
+
     if (!object_root) {
       object_root = object_preview;
       object_root.set_hide_on_close(true);
       object_root.connect("close-request", () => {
         onWindowChange(false);
       });
+    } else {
+      for (const prop of object_root.constructor.list_properties()) {
+        if (!(prop.flags & GObject.ParamFlags.WRITABLE)) continue;
+        if (!(prop.flags & GObject.ParamFlags.READABLE)) continue;
+        if (prop.flags & GObject.ParamFlags.CONSTRUCT_ONLY) continue;
+
+        const prop_name = prop.get_name();
+        if (
+          [
+            // The new window does not have "csd" at this time
+            "css-classes",
+            // Triggers
+            // Gtk-CRITICAL Widget of type “GtkApplicationWindow” already has an accessible role of type “GTK_ACCESSIBLE_ROLE_WIDGET”
+            "accessible-role",
+            // These properties override current window size
+            "default-width",
+            "default-height",
+          ].includes(prop_name)
+        ) {
+          continue;
+        }
+
+        const new_value = object_preview[prop_name];
+        if (new_value instanceof Gtk.Widget) {
+          // Remove parent widget to prevent
+          // Can't set new parent  0x5649879daa40 on widget GtkHeaderBar 0x564987790d90, which already has parent GtkApplicationWindow 0x564988f31a40
+          object_preview[prop_name] = null;
+        }
+
+        const old_value = object_root[prop_name];
+        if (!old_value !== new_value) {
+          object_root[prop_name] = new_value;
+        }
+      }
     }
-    adoptChild(object_preview, object_root);
```

Comment:
since `child` and `content` are properties, and we now update all properties, we don't need this anymore